django_fido/management/commands/download_authenticator_metadata.py

```diff
@@ -0,0 +1,88 @@
+"""Command to download metadata for authenticators."""
+from __future__ import unicode_literals
+
+import json
+from base64 import urlsafe_b64decode
+
+import requests
+from cryptography.hazmat.backends import default_backend
+from cryptography.x509 import load_pem_x509_certificate, ExtensionNotFound
+from cryptography.x509.oid import ExtensionOID
+from django.core.management.base import BaseCommand, CommandError
+from jwcrypto.jwk import JWK
+from jwcrypto.jwt import JWT
+
+from django_fido.models import AuthenticatorMetadata
+from django_fido.settings import SETTINGS
+
+PEM_CERT_TEMPLATE = '-----BEGIN CERTIFICATE-----\n{}\n-----END CERTIFICATE-----\n'
+
+
+class Command(BaseCommand):
+    """Download metadata for authenticators."""
+
+    help = "Download metadata for authenticators."""
+
+    def add_arguments(self, parser):
+        """Parse command arguments."""
+
+    def handle(self, **options):
+        """Donwload and parse metadata from metadata service."""
+        if SETTINGS.metadata_service is None:
+            raise CommandError('mds_access_token setting must be specified for this command to work.')
+        metadata = requests.get(SETTINGS.metadata_service['mds_url'],
+                                params={'token': SETTINGS.metadata_service['mds_access_token']})
+        # First, we decode the unverified headers to get the certificate
+        decoded_jwt = JWT(jwt=metadata.content.decode())
+        # x5c element in header contains the signing certificate and possibly intermediate certificates
+        # Use the first one to verify signature, the others are used to verify the first one
+        # TODO: Use x5c[1:] to verify the chain to key
+        # TODO: Use this root certificate to verify the chain of trust ...
+        # http://www.yothenberg.com/validate-x509-certificate-in-python/
+        if SETTINGS.metadata_service['cert_file'] is not None:
+            with open(SETTINGS.metadata_service.cert_file, 'rb') as cert_file:
+                key = JWK.from_pem(cert_file.read())
+        decoding_key = JWK.from_pem(PEM_CERT_TEMPLATE.format(decoded_jwt.token.jose_header['x5c'][0]).encode())
```

Comment:
`PEM_CERT_TEMPLATE` looks quite ugly.

Reply:
I couldn't find anything better as it is PEM formatted cert and jwcrypto only reads from a PEM file content.